training/distributed_training/pytorch/data_parallel/efficientnet/EfficientNet_Notebook.ipynb

```diff
@@ -0,0 +1,492 @@
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "00bc6fe9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Configure FSx Input for your SageMaker Training job\n",
+    "from sagemaker.inputs import FileSystemInput\n",
+    "\n",
+    "# FSx file system ID with your training dataset. Example: 'fs-0bYYYYYY'\n",
+    "file_system_id = \"<fsx_id>\"\n",
+    "# FSx path with your training data # Example: '/fsx_mount_name/imagenet'\n",
+    "file_system_directory_path = \"/<fsx_mount_name>/<path_to_data>\"\n",
+    "file_system_access_mode = \"rw\"\n",
+    "file_system_type = \"FSxLustre\"\n",
+    "train_fs = FileSystemInput(\n",
+    "    file_system_id=file_system_id,\n",
+    "    file_system_type=file_system_type,\n",
+    "    directory_path=file_system_directory_path,\n",
+    "    file_system_access_mode=file_system_access_mode,\n",
+    ")\n",
+    "# Specify the training data channel using the FSx filesystem. This will be provided to the SageMaker training job later\n",
+    "data_channels = {\"train\": train_fs}\n",
+    "\n",
+    "# The following variables will be used later in the notebook\n",
+    "fsx_subnets = [\"<subnet_id>\"]  # Should be the subnet used for FSx. Example: subnet-0f9XXXX\n",
+    "fsx_security_group_ids = [\n",
+    "    \"<security_group_id>\"\n",
+    "]  # Should be the security group used for FSx. sg-03ZZZZZZ"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "085c55c1",
+   "metadata": {},
+   "source": [
+    "## Prepare SageMaker Training Images\n",
+    "\n",
+    "SageMaker by default uses the latest [Amazon Deep Learning Container Images (DLC)](https://github.com/aws/deep-learning-containers/blob/master/available_images.md) PyTorch training image. In this step, we use it as a base image and install additional dependencies required for training EfficientNet model. Run the cells below to assign a name for your image and observe the dockerfile."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "84aceb16",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "image = (\n",
+    "    \"pt-smdataparallel-efficientnet-sagemaker\"  # Example: pt-smdataparallel-efficientnet-sagemaker\n",
+    ")\n",
+    "tag = \"latest\"  # Example: latest"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0b1ee180",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pygmentize ./Dockerfile"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5924dc79",
+   "metadata": {},
+   "source": [
+    "### Get the training script\n",
+    "In this [GitHub repository](https://github.com/HerringForks/SMDDP-Examples/tree/main/pytorch/image_classification), we have forked an EfficientNet example from [NVIDIA/DeepLearningExamples](https://github.com/NVIDIA/DeepLearningExamples/tree/master/PyTorch/Classification/ConvNets) and adapted the training script to work with SMDDP. Note that the only required changes are:\n",
```

Comment:
let's emphatically say here that adapting to use SMDDP only requires 3 (or whatever number) lines of code change. and give pointer to documentation with new interface.

Reply:
Sounds good, I will do that